pppd/chap-md5.c

```diff
@@ -63,27 +63,43 @@ chap_md5_verify_response(int id, char *name,
 			 unsigned char *challenge, unsigned char *response,
 			 char *message, int message_space)
 {
-	MD5_CTX ctx;
 	unsigned char idbyte = id;
 	unsigned char hash[MD5_HASH_SIZE];
+    unsigned int  hash_len = MD5_HASH_SIZE;
 	int challenge_len, response_len;
+    bool success = 0;
 
 	challenge_len = *challenge++;
 	response_len = *response++;
 	if (response_len == MD5_HASH_SIZE) {
+
 		/* Generate hash of ID, secret, challenge */
-		MD5_Init(&ctx);
-		MD5_Update(&ctx, &idbyte, 1);
-		MD5_Update(&ctx, secret, secret_len);
-		MD5_Update(&ctx, challenge, challenge_len);
-		MD5_Final(hash, &ctx);
-
-		/* Test if our hash matches the peer's response */
-		if (memcmp(hash, response, MD5_HASH_SIZE) == 0) {
-			slprintf(message, message_space, "Access granted");
-			return 1;
-		}
+        PPP_MD_CTX* ctx = PPP_MD_CTX_new();
+        if (ctx) {
+
+            if (PPP_DigestInit(ctx, PPP_md5())) {
```

Comment:
Whitespace is inconsistent here (and the blank lines look a bit odd IMO)

Reply:
I'll fix up the white space inconsistencies

Reply:
There is a case for disabling the include of openssl (embedded developers). Also, openssl can be configured not to include certain encryption algorithms and thus we would need to provide our own. Then there is the few other projects (I would guess embedded, that would want to use something else than openssl). The API is modeled after openssl and can easily be search and replaced. So there is that ...